packit_service/worker/jobs.py

```diff
@@ -212,97 +91,77 @@ def get_handlers_for_check_rerun(check_name_job: str) -> Set[Type[JobHandler]]:
+        return self._service_config
 
-    return matching_jobs
+    @staticmethod
+    def process_message(event: dict) -> List[TaskResults]:
+        """
+        Entrypoint for message processing.
 
+        Args:
+            event:  dict with webhook/fed-mes payload
 
-def push_initial_metrics(
-    task_accepted_time: datetime,
-    event: Event,
-    handler: JobHandler,
-    number_of_build_targets: Optional[int] = None,
-):
-    pushgateway = Pushgateway()
-    response_time = measure_time(end=task_accepted_time, begin=event.created_at)
-    logger.debug(f"Reporting initial status time: {response_time} seconds.")
-    pushgateway.initial_status_time.observe(response_time)
-    if response_time > 15:
-        pushgateway.no_status_after_15_s.inc()
+        Returns:
+            List of results of the processing tasks.
+        """
+        event_object: Any = Parser.parse_event(event)
 
-    # set the time when the accepted status was set so that we can use it later for measurements
-    event.task_accepted_time = task_accepted_time
+        if not (event_object and event_object.pre_check()):
+            return []
 
-    if isinstance(handler, CoprBuildHandler) and number_of_build_targets:
-        for _ in range(number_of_build_targets):
-            pushgateway.copr_builds_queued.inc()
+        return SteveJobs(event_object).process()
 
-    pushgateway.push()
+    def process(self) -> List[TaskResults]:
+        if not self.is_project_public_or_enabled_private():
+            return []
 
+        processing_results = None
 
-class SteveJobs:
-    """
-    Steve makes sure all the jobs are done with precision.
-    """
+        # installation is handled differently b/c app is installed to GitHub account
+        # not repository, so package config with jobs is missing
+        if isinstance(self.event, InstallationEvent):
+            GithubAppInstallationHandler.get_signature(
+                event=self.event, job=None
+            ).apply_async()
+        else:
+            # Processing the jobs from the config.
+            processing_results = self.process_jobs()
 
-    def __init__(self):
-        self._service_config = None
-        log_job_versions()
+        if processing_results is None:
+            processing_results = [
+                TaskResults.create_from(
+                    success=True,
+                    msg="Job created.",
+                    job_config=None,
+                    event=self.event,
+                )
+            ]
 
-    @property
-    def service_config(self):
-        if self._service_config is None:
-            self._service_config = ServiceConfig.get_service_config()
-        return self._service_config
+        return processing_results
 
-    def report_task_accepted(
-        self, event: Event, handler: JobHandler, job_config: JobConfig
-    ):
+    def report_task_accepted(self, handler: JobHandler, job_config: JobConfig):
```

Comment:
This method is still pretty long. Can we at least separate creation of `job_helper` or something like this?
I was thinking about something like this in separate method
```py
...

params = {
    service_config=self.service_config,
    package_config=self.event.package_config,
    project=self.event.project,
    metadata=EventData.from_event_dict(self.event.get_dict()),
    db_trigger=self.event.db_trigger,
    job_config=job_config,
}

if propose_downstream:
    other_params = {branches_override=self.event.branches_override}
else:
    other_params = {...}

return helper_cls(params | other_params)
```

Reply:
Good point! I split it, PTAL